Mathlib/Data/Finset/LocallyFinite/Box.lean

```diff
@@ -48,6 +48,16 @@ lemma box_succ_disjUnion (n : ℕ) :
 
 @[simp] lemma zero_mem_box : (0 : α) ∈ box n ↔ n = 0 := by cases n <;> simp [box_succ_eq_sdiff]
 
+lemma mem_box_eq_zero_iff_eq_zero (x : α) (hx : x ∈ box n) : x = 0 ↔ n = 0 := by
```

Comment:
Should `x` be implicit?

Would `eq_zero_iff_eq_zero_of_mem_box` be a more consistent name?